server/base/src/test/java/org/apache/accumulo/server/manager/state/TabletManagementIteratorTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.manager.state;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.junit.jupiter.api.Test;
+
+public class TabletManagementIteratorTest {
+
+  @Test
+  public void testRanges() throws IOException {
+    TabletManagementIterator iter = new TabletManagementIterator();
+
+    // We don't call init, so expect a IllegalStateException on success and
+    // and IllegalArgumentException on failure
+    Key goodStartKey = new Key("row");
+    Key goodEndKey = new Key("rowEnd");
+
+    Key badStartKey = new Key("row", "colf", "colq", 1234L);
+    Key badEndKey = new Key("rowEnd", "colf", "colq", 1234L);
+
+    assertThrows(IllegalStateException.class, () -> iter.seek(null, Set.of(), false));
+    assertThrows(IllegalStateException.class,
+        () -> iter.seek(new Range((Key) null, (Key) null), Set.of(), false));
+    assertThrows(IllegalStateException.class,
+        () -> iter.seek(new Range(goodStartKey, goodEndKey), Set.of(), false));
+    assertThrows(IllegalArgumentException.class,
+        () -> iter.seek(new Range(goodStartKey, badEndKey), Set.of(), false));
```

Comment:
Could checks the message to make sure the IllegalArgumentException is coming from the expected place and not something like Range constructor.

```suggestion
        () -> iter.seek(new Range(goodStartKey, badEndKey), Set.of(), false)).getMessage().contains("TabletManagementIterator must be seeked");
```

Reply:
Interesting, I didn't realize that the `assertThrows` method actually returns the Exception. I added your suggestion to each of the tests that should fail in 6659ab3, but also wrapped them in an `assertTrue`.